compiler/src/main/kotlin/io/spine/protodata/renderer/SourceFile.kt

```diff
@@ -281,6 +247,38 @@ internal constructor(
     }
 }
 
+/**
+ * A splitter to divide code fragments into lines.
+ *
+ * Uses a regular expression to match line breaks, with or without carriage returns.
+ */
+public val lineSplitter: Splitter = Splitter.on(Pattern.compile("\r?\n"))
+
+/**
+ * Creates a new fluent builder for adding code at the given [insertionPoint].
+ *
+ * If the [insertionPoint] is not found in the code, no action will be performed as the result.
+ * If there are more than one instances of the same insertion point, the code will be added to
+ * all of them.
+ *
+ * Insertion points should be marked with comments of special format. The added code is always
+ * inserted after the line with the comment, and the line with the comment is preserved.
+ */
+public fun SourceFile.at(insertionPoint: InsertionPoint): SourceAtPoint =
```

Comment:
Regarding `at(..)` and `lines()`, I think you should reconsider. Those are not auxiliary functions, but the primary API any SPI-user will be using. If there are some formal requirements regarding what is considered "auxiliary" and what is not, let's see if it should be changed or if we should suppress these particular cases.